errors/error.go

```diff
@@ -1,6 +1,17 @@
 package errors
 
+import "github.com/docker/distribution/registry/api/errcode"
+
 // This file contains all of the errors that can be generated from the
 // docker engine but are not tied to any specific top-level component.
 
 const errGroup = "engine"
+
+// IsErr returns a bool indicating if the passed in error matches the expected error
+func IsErr(err error, expected errcode.ErrorCode) bool {
```

Comment:
this is a really bad func name if you think about it ;)

maybe something more descriptive but seeing that this is not used in this PR i would remove this file from the PR.